src/NServiceBus.Core/Pipeline/BehaviorContext.cs

```diff
@@ -1,27 +1,20 @@
 namespace NServiceBus
 {
     using System;
-    using System.Diagnostics.CodeAnalysis;
     using System.Threading;
     using Extensibility;
     using Pipeline;
 
-    class BehaviorContext : ContextBag, IBehaviorContext
+    abstract class BehaviorContext : ContextBag, IBehaviorContext
     {
-        protected BehaviorContext(IBehaviorContext parentContext) : this(parentContext, parentContext?.CancellationToken ?? default)
-        { }
+        //TODO can parent ever be null here?
+        protected BehaviorContext(IBehaviorContext parentContext) : this(parentContext.Extensions, parentContext.CancellationToken)
+        {
+        }
 
-        [SuppressMessage("Code", "PS0014:Methods should not have both CancellationToken parameters and parameters implementing ICancellableContext", Justification = "The cancellation tokens are linked together if required.")]
-        public BehaviorContext(IBehaviorContext parentContext, CancellationToken cancellationToken = default) : base(parentContext?.Extensions)
+        public BehaviorContext(ContextBag parentContext, CancellationToken cancellationToken = default) : base(parentContext)
         {
-            if (parentContext != null && (cancellationToken == default || cancellationToken == parentContext.CancellationToken))
-            {
-                CancellationToken = parentContext.CancellationToken;
-            }
-            else
-            {
-                CancellationToken = cancellationToken;
-            }
```

Comment:
I'm not sure whether the new implementation really covers the different combinations properly so if somebody has some spare brain cycles, please look at this change more closely.

Reply:
Puh I need to think this through with another lool